samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java

```diff
@@ -202,9 +209,16 @@ private UdfResolver createUdfResolver(Map<String, String> config) {
 
     Collection<RelRoot> relRoots = dslConverter.convertDsl(String.join("\n", dslStmts));
 
-    for (RelRoot relRoot : relRoots) {
-      SamzaSqlApplicationConfig.populateSystemStreams(relRoot.project(), inputSystemStreams, outputSystemStreams);
-    }
+    // the snippet below dose not work when sql is a query
+    // for (RelRoot relRoot : relRoots) {
+    //   SamzaSqlApplicationConfig.populateSystemStreams(relRoot.project(), inputSystemStreams, outputSystemStreams);
+    // }
```

Comment:
Is there no way around commenting these ? In the case of DSLs, we won't be able to parse the query and get the input/output streams. All we have is relational plan in the form of RelRoot.

Reply:
Before QueryTranslator performs "translate()", "SamzaSqlApplicationConfig" or "SamzaSqlApplication" has to know if the Sql statements has queries . If we still use this approach that passes the query only to Calcite if users perform SELECT sql statement. To around commenting these lines, we could add an API "hasSqlQuery()" in "interface DslConverter" or somewhere. 

Or, we can think of other solutions, for example, we still pass the whole Sql statement to Calcite, but still we need to figure out a way to deal with the issue of missing schema of log.outputStream.